packages/owl-bot/src/config-files.ts

```diff
@@ -67,45 +67,89 @@ export interface OwlBotYaml {
 // The default path where .OwlBot.yaml is expected to be found.
 export const DEFAULT_OWL_BOT_YAML_PATH = '.github/.OwlBot.yaml';
 
-function validatePath(path: string, fieldName: string) {
-  if (path && path[0] !== '/') {
-    throw `I expected the first character of ${fieldName} to be a '/'.  Instead, I found ${path}.`;
+function validatePath(
+  path: string,
+  fieldName: string,
+  errorMessages: string[]
+) {
+  if (!path.startsWith('/')) {
+    errorMessages.push(
+      `I expected the first character of ${fieldName} to be a '/'.  Instead, I found ${path}.`
+    );
   }
 }
 
-// Throws an exception if the object does not have the necessary structure.
-// Otherwise, returns the same object as an OwlBotYaml.
-export function owlBotYamlFromText(yamlText: string): OwlBotYaml {
-  const o = yaml.load(yamlText) ?? {};
-  const validate = new Ajv().compile(owlBotYamlSchema);
-  if (validate(o)) {
-    const yaml = o as OwlBotYaml;
-    for (const deepCopy of yaml['deep-copy-regex'] ?? []) {
-      validatePath(deepCopy.dest, 'dest');
-      validatePath(deepCopy.source, 'source');
-      // Confirm it's a valid regular expression.
-      toFrontMatchRegExp(deepCopy.source);
-    }
-    for (const removePath of yaml['deep-remove-regex'] ?? []) {
-      validatePath(removePath, 'deep-remove-regex');
-      // Confirm it's a valid regular expression.
-      toFrontMatchRegExp(removePath);
+function validateRegex(regex: string, errorMessages: string[]) {
```

Comment:
Please add a /** */ comment for this function.

Reply:
Done